apps/settings/sub_menu/about_controller.cpp

```diff
@@ -113,10 +113,22 @@ bool AboutController::handleEvent(Ion::Events::Event event) {
         MessageTableCellWithBuffer * myCell = (MessageTableCellWithBuffer *)m_selectableTableView.selectedCell();
         char batteryLevel[5];
         if(strchr(myCell->accessoryText(), '%') == NULL){
-          int batteryLen = Poincare::Integer((int) ((Ion::Battery::voltage() - 3.6) * 166)).serialize(batteryLevel, 5);
-          batteryLevel[batteryLen] = '%';
-          batteryLevel[batteryLen+1] = '\0';
-        }else{
+            if((int) ((Ion::Battery::voltage() - 3.6) * 166) < 0) {
+              batteryLevel[0] = '0';
+              batteryLevel[1] = '%';
+              batteryLevel[2] = '\0';
+            }
+            else
+            {
+              int batteryLen = Poincare::Integer((int) ((Ion::Battery::voltage() - 3.6) * 166)).serialize(batteryLevel, 5);
+              batteryLevel[batteryLen] = '%';
+              batteryLevel[batteryLen+1] = '\0';
+            }
+            if(Ion::Battery::isCharging() && (int) ((Ion::Battery::voltage() - 3.6) * 166) >= 100) {
+              myCell->setAccessoryText("100%");
+              return true;
+            }
```

Comment:
```suggestion
          double voltage = (Ion::Battery::voltage() - 3.6) * 166;
          if((voltage < 0.0) {
            myCell->setAccessoryText("1%"); // We cheat...
            return true;
          } else if (voltage >= 100.0) {
            myCell->setAccessoryText("100%");
            return true;
          } else {
            int batteryLen = Poincare::Integer((int) voltage).serialize(batteryLevel, 5);
            batteryLevel[batteryLen] = '%';
            batteryLevel[batteryLen+1] = '\0';
          }
```

Reply:
yes it is better but you should use float type instead of double for memory usage x)x)x)x)